ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -577,26 +675,117 @@ private object OracleQueries extends Queries {
       pvs: Put[Vector[String]],
   ): T[Query0[DBContract[Mark, JsValue, JsValue, Vector[String]]]] = {
     val Seq((tpid, predicate @ _)) = queries // TODO SC handle more than one
-    val _ = parties
     println("selecting")
-    val q = sql"""SELECT contract_id, key, payload, agreement_text
-                  FROM contract
-                  WHERE tpid = $tpid""" // TODO SC AND (""" ++ predicate ++ sql")"
+    import Queries.CompatImplicits.catsReducibleFromFoldable1
+    // % is explicitly reserved by specification as a delimiter
+    val q = sql"""SELECT c.contract_id, key, payload, agreement_text, sd.parties, od.parties
```

Comment:
maybe add a todo to clean up this query. We should probably either try to switch to arrays or try to only do one join per table.